src/libawkward/array/IndexedArray.cpp

```diff
@@ -2215,14 +2226,15 @@ namespace awkward {
                                               stable,
                                               keepdims);
 
-    Index64 nextoutindex(index_.length());
+    Index64 nextoutindex(parents_length);
```

Comment:
I take it this was the significant change, right?

Reply:
it looks like the length of parents and index are equal in all tested cases, but I changed it to be more explicit because the iteration is on parents